### DISCUSSION TITLE
Adding further replay analysis tools

### DISCUSSION THREAD

**Discussion by minisbett** [ppy/osu#29692] [Ideas]
As a followup for the https://github.com/ppy/osu/pull/27334 PR that's been worked on and asking about it in chat, I'd like to propose more replay information to be added to such a feature.

Namingly, there is some statistics that have been proven to be useful for replay analysis:
- Hold time distribution (amount of milliseconds a key is pressed down, both combined and per-key, ideally also separated between circles and sliders by assuming the object closest to the press)
- Press time distribution (amount of milliseconds offset from the nearest object, also separated like mentioned above)
- A frametime graph, displaying the milliseconds between replay frames

All of the above can be visualized using some kind of graph, so I believe theoretically existing UI components can be reused, though I'm unsure how fitting it'd look.

What would also be HUGE for analysis would be a visualization of key presses in a timeline-fashion, similar to the Rewind 3rd-party tool. A screenshot can be found below.
![image](https://github.com/user-attachments/assets/be6b82c7-a94f-4e01-a69d-c50f3fe6f97d)

I'm not sure how these can be brought in without looking too overwhelming since from what I can see the replay analysis currently being implemented is not an extra mode, but just some overlay when viewing a replay the normal way. Maybe there's a way to add a button to the "Analysis Settings" settings panel that opens some sidebar or anything the-like.

**Comment by mixalisnew** [comment]
i think the default for the replays should be as it is and for players who want to analyze the replay a Advanced replay button maybe. Also a toggle for which would be default from the settings is a good idea.

**Reply by minisbett** [reply]
From what I saw in peppy's visual design pass stream there's an extra category at the right side when viewing replays where you can toggle certain overlays (hit makers, aim line etc.). So yeah my idea would be allowing access to those metrics somewhere there so the average user doesn't accidentally stumble on it and is confused, as it certainly is more advanced information being presented.

**Reply by mixalisnew** [reply]
Sounds good
